File: backend/src/TarlaAsistani.Domain/Enums/MarketCategory.cs
namespace TarlaAsistani.Domain.Enums;

/// <summary>
/// Tarımsal piyasa verilerinin kategorilerini temsil eder.
/// </summary>
public enum MarketCategory
{
    /// <summary>
    /// Akaryakıt ürünleri (Motorin, Benzin vb.).
    /// </summary>
    Fuel = 1,

    /// <summary>
    /// Gübre ürünleri (Üre, DAP vb.).
    /// </summary>
    Fertilizer = 2,

    /// <summary>
    /// Mahsul ve hububat ürünleri (Ekmeklik Buğday, Mısır vb.).
    /// </summary>
    Crop = 3,

    /// <summary>
    /// Döviz kurları (USD/TRY, EUR/TRY vb.).
    /// </summary>
    Fx = 4
}

File: mobile/lib/features/market/domain/models/market_category.dart
/// Tarımsal piyasa verilerinin kategorilerini temsil eden enum.
enum MarketCategory {
  /// Akaryakıt ürünleri (Motorin, Benzin vb.)
  fuel,

  /// Gübre ürünleri (Üre, DAP vb.)
  fertilizer,

  /// Mahsul ve hububat ürünleri (Buğday, Mısır vb.)
  crop,

  /// Döviz kurları (USD, EUR vb.)
  fx,

  /// Tüm kategorileri kapsayan filtre seçeneği
  all;

  /// Kullanıcı arayüzünde gösterilecek Türkçe etiket.
  String get displayName => switch (this) {
        MarketCategory.fuel => 'Akaryakıt',
        MarketCategory.fertilizer => 'Gübre',
        MarketCategory.crop => 'Mahsul',
        MarketCategory.fx => 'Döviz',
        MarketCategory.all => 'Tümü',
      };

  /// REST API sorgularında kullanılan küçük harfli parametre değeri.
  String get apiValue => switch (this) {
        MarketCategory.fuel => 'fuel',
        MarketCategory.fertilizer => 'fertilizer',
        MarketCategory.crop => 'crop',
        MarketCategory.fx => 'fx',
        MarketCategory.all => 'all',
      };

  /// API'den gelen dize değerini büyük/küçük harf duyarsız eşleştirerek enum değerine dönüştürür.
  factory MarketCategory.fromApiValue(String? value) {
    if (value == null || value.trim().isEmpty) {
      return MarketCategory.all;
    }

    final normalized = value.trim().toLowerCase();
    return MarketCategory.values.firstWhere(
      (c) => c.apiValue == normalized || c.name.toLowerCase() == normalized,
      orElse: () => MarketCategory.all,
    );
  }
}
